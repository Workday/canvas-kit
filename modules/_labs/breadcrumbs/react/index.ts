import Breadcrumbs from './lib/Breadcrumbs';

export default Breadcrumbs;
export {Breadcrumbs};
export * from './lib/Breadcrumbs';
