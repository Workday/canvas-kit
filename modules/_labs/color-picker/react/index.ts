export {default, default as ColorPicker, ColorPickerProps} from './lib/ColorPicker';
export {Swatch} from './lib/parts/Swatch';
