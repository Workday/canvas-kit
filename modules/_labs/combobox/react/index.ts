import Combobox from './lib/Combobox';

export default Combobox;
export {Combobox};
export * from './lib/Combobox';
