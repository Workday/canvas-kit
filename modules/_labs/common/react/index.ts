export * from './lib/Flex';
export * from './lib/Box';
