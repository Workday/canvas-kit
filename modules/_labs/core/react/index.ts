import type from './lib/type';
import space from './lib/space';
export default type;
export {type, space};
export * from './lib/type';
export * from './lib/StaticStates';
