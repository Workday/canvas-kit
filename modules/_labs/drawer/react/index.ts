import Drawer from './lib/Drawer';
import DrawerHeader from './lib/DrawerHeader';
export {Drawer, DrawerHeader};
export default Drawer;
export * from './lib/Drawer';
