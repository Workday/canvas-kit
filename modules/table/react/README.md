# Canvas Kit Table

A component for applying Canvas styling to table elements.

## Installation

```sh
yarn add @workday/canvas-kit-react
```

or

```sh
yarn add @workday/canvas-kit-react-table
```

## Usage

Extends `<table>` with Canvas data table styling. Use with native `<thead>` and `<tbody>` elements
for respective styling.

Use [`<TableRow>`](#TableRow) component instead of `<tr>` to define rows.

```tsx
import {Table, TableRow} from '@workday/canvas-kit-react-table';

<Table>
  <thead>
    <TableRow header={true}>
      <th>ID</th>
      <th>Name</th>
    </TableRow>
  </thead>
  <tbody>
    <TableRow>
      <td>1</td>
      <td>Aidan Brown</td>
    </TableRow>
    <TableRow>
      <td>2</td>
      <td>Betty Chen</td>
    </TableRow>
  </tbody>
</Table>;
```

## Static Properties

> None

## Component Props

### Required

> None

### Optional

> None

# TableRow

## Usage

Extends `<tr>` with row states and styling. Use with `<th>` and `<td>` elements.

```tsx
import {Table, TableRow} from '@workday/canvas-kit-react-table';

<Table>
  <thead>
    <TableRow header={true}>
      <th>ID</th>
      <th>Name</th>
    </TableRow>
  </thead>
  <tbody>
    <TableRow state={TableRow.State.Error}>
      <td>1</td>
      <td>Aidan Brown</td>
    </TableRow>
    <TableRow state={TableRow.State.Alert}>
      <td>2</td>
      <td>Betty Chen</td>
    </TableRow>
  </tbody>
</Table>;
```

## Static Properties

#### `State: TableRowState`

```tsx
<TableRow state={TableRow.State.Error}>
```

| Theme        | Description                              |
| ------------ | ---------------------------------------- |
| `Error`      | Red border, light red background         |
| `InputError` | Regular borders, light red background    |
| `Alert`      | Orange border, light orange background   |
| `InputAlert` | Regular borders, light orange background |
| `Hover`      | Grey background                          |
| `Selected`   | Blue border, light blue background       |

`InputError` and `InputAlert` are used for cases when the error is on the input-level rather than
the entire row. This style produces a colored row without a darkened border.

## Component Props

### Optional

#### `state: TableRowState`

> The state of the row

Default: `null`

---

#### `header: boolean`

> Whether or not the row contains header elements. If true, special styling will be applied for
> header elements.

Default: `false`
