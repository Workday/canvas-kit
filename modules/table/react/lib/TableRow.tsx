import * as React from 'react';
import {Interpolation} from 'emotion';
import styled from 'react-emotion';
import {rgba} from 'polished';
import {colors, spacing, spacingNumbers} from '@workday/canvas-kit-react-core';
import {borderColor, borderWidth, cellBorder} from './Table';

export enum TableRowStates {
  Error,
  Alert,
  InputError,
  InputAlert,
  Hover,
  Selected,
}

export interface TableRowProps extends React.HTMLAttributes<HTMLTableRowElement> {
  /**
   * State of the row
   */
  state?: TableRowStates;

  /**
   * Whether or not the row contains header elements
   */
  header?: boolean;
}

const errorColor = colors.cinnamon400;
const errorColorLight = colors.cinnamon200;
const alertColor = colors.cantaloupe400;
const alertColorLight = colors.cantaloupe200;

function makeColoredRow(_bgColor: string, _borderColor: string): Interpolation {
  return {
    backgroundColor: _bgColor,
    position: 'relative',

    // Bottom border
    '&:before, &:after': {
      position: 'absolute',
      left: 0,
      top: -borderWidth,
      content: '""',
      width: `calc(100% + ${borderWidth}px)`,
      height: borderWidth,
      backgroundColor: _borderColor,
    },

    '&:after': {
      top: 'auto',
      bottom: -borderWidth,
    },

    '&:first-child': {
      boxShadow: `inset ${borderWidth}px 0 0 ${_borderColor}`,
    },

    '&:last-child': {
      boxShadow: `inset -${borderWidth}px 0 0 ${_borderColor}`,
      '&:before, &:after': {
        width: '100%',
      },
    },
  };
}

function makeColoredRowStyle(_bgColor: string, _borderColor: string): Interpolation {
  const lightenedBg = rgba(_bgColor, 0.2);
  return {
    'th, td': [
      makeColoredRow(lightenedBg, _bgColor),
      {
        borderLeftColor: _borderColor,
        borderRightColor: _borderColor,
      },
    ],
    '&:hover': {
      'th, td': {
        backgroundColor: lightenedBg,
      },
    },
  };
}

function makeBorderlessStyle(_bgColor: string): Interpolation {
  return {
    'th, td': {
      backgroundColor: rgba(_bgColor, 0.2),
    },
  };
}

const Row = styled('tr')<TableRowProps>(
  {
    'th, td': {
      backgroundColor: colors.frenchVanilla100,
      padding: `${spacing.xs} ${spacing.xxs}`,
      fontSize: 13,
      borderRight: cellBorder,
      borderBottom: cellBorder,
      height: spacingNumbers.xl + spacingNumbers.xxs,
      boxSizing: 'border-box',
      transition: 'background-color 0.2s',
    },
    td: {
      '&:last-child': {
        borderRight: 'none',
      },
    },
    th: {
      height: spacing.xxl,
      '&:last-child': {
        borderRight: 'none',
      },
    },
    '&:last-child': {
      'th, td': {
        borderBottom: 'none',
      },
    },
  },
  ({header}) =>
    header && {
      '&:last-child': {
        'th, td': {
          borderBottom: '1px solid transparent',
        },
      },
      'th, td': {
        position: 'relative',
        backgroundColor: colors.frenchVanilla100,
        verticalAlign: 'middle', // Bottom border
        '&:after': {
          position: 'absolute',
          left: 0,
          bottom: -borderWidth,
          content: '""',
          width: `calc(100% + ${borderWidth}px)`,
          height: borderWidth,
          backgroundColor: borderColor,
        },
        '&:last-child': {
          '&:after': {
            width: '100%',
          },
        },
      },
    },
  ({state}) => {
    const styles: Interpolation = [];

    switch (state) {
      case TableRowStates.InputError:
        styles.push(makeBorderlessStyle(errorColor));
        break;
      case TableRowStates.Error:
        styles.push(makeColoredRowStyle(errorColor, errorColorLight));
        break;
      case TableRowStates.InputAlert:
        styles.push(makeBorderlessStyle(alertColor));
        break;
      case TableRowStates.Alert:
        styles.push(makeColoredRowStyle(alertColor, alertColorLight));
        break;
      case TableRowStates.Selected:
        styles.push({
          'th, td': [
            makeColoredRow(colors.blueberry100, colors.blueberry500),
            {
              '&:after': {
                zIndex: 2,
              },
            },
          ],
        });
        break;
      default:
        styles.push({
          '&:hover': {
            'th, td': {
              backgroundColor: colors.soap300,
            },
          },
          '&:focus': {
            outline: 'none',
            'th, td': [
              makeColoredRow(colors.blueberry100, colors.blueberry500),
              {
                '&:after': {
                  zIndex: 2,
                },
              },
            ],
          },
        });
    }

    return styles;
  }
);

export default class TableRow extends React.Component<TableRowProps> {
  public static State = TableRowStates;

  public render() {
    const {state, header, children, ...elemProps} = this.props;

    return (
      <Row state={state} header={header} {...elemProps}>
        {children}
      </Row>
    );
  }
}
