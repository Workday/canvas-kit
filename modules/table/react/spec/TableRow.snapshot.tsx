import * as React from 'react';
import * as renderer from 'react-test-renderer';
import TableRow, {TableRowStates} from '../lib/TableRow';

describe('Table Row Snapshots', () => {
  test('renders as expected', () => {
    const component = renderer.create(
      <TableRow>
        <td>cell</td>
      </TableRow>
    );
    expect(component).toMatchSnapshot();
  });

  test('header mode', () => {
    const component = renderer.create(
      <TableRow header={true}>
        <th>cell</th>
      </TableRow>
    );
    expect(component).toMatchSnapshot();
  });

  describe('states', () => {
    function createStateComponent(state: TableRowStates) {
      return renderer.create(
        <TableRow state={state}>
          <td>cell</td>
        </TableRow>
      );
    }
    test('Error', () => {
      const component = createStateComponent(TableRowStates.Error);
      expect(component).toMatchSnapshot();
    });

    test('Alert', () => {
      const component = createStateComponent(TableRowStates.Alert);
      expect(component).toMatchSnapshot();
    });

    test('Input Error', () => {
      const component = createStateComponent(TableRowStates.InputError);
      expect(component).toMatchSnapshot();
    });

    test('Input Alert', () => {
      const component = createStateComponent(TableRowStates.InputAlert);
      expect(component).toMatchSnapshot();
    });

    test('Hover', () => {
      const component = createStateComponent(TableRowStates.Hover);
      expect(component).toMatchSnapshot();
    });

    test('Selected', () => {
      const component = createStateComponent(TableRowStates.Selected);
      expect(component).toMatchSnapshot();
    });
  });
});
