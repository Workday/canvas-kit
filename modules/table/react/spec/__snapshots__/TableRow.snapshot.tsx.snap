// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table Row Snapshots header mode 1`] = `
.emotion-0 th,
.emotion-0 td {
  background-color: #ffffff;
  padding: 12px 8px;
  font-size: 13px;
  border-right: 1px solid #dfe2e6;
  border-bottom: 1px solid #dfe2e6;
  height: 48px;
  box-sizing: border-box;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.emotion-0 td:last-child {
  border-right: none;
}

.emotion-0 th {
  height: 64px;
}

.emotion-0 th:last-child {
  border-right: none;
}

.emotion-0:last-child th,
.emotion-0:last-child td {
  border-bottom: none;
}

.emotion-0:last-child th,
.emotion-0:last-child td {
  border-bottom: 1px solid transparent;
}

.emotion-0 th,
.emotion-0 td {
  position: relative;
  background-color: #ffffff;
  vertical-align: middle;
}

.emotion-0 th:after,
.emotion-0 td:after {
  position: absolute;
  left: 0;
  bottom: -1px;
  content: "";
  width: calc(100% + 1px);
  height: 1px;
  background-color: #dfe2e6;
}

.emotion-0 th:last-child:after,
.emotion-0 td:last-child:after {
  width: 100%;
}

.emotion-0:hover th,
.emotion-0:hover td {
  background-color: #e8ebed;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:focus th,
.emotion-0:focus td {
  background-color: #d7eafc;
  position: relative;
}

.emotion-0:focus th:before,
.emotion-0:focus td:before,
.emotion-0:focus th:after,
.emotion-0:focus td:after {
  position: absolute;
  left: 0;
  top: -1px;
  content: "";
  width: calc(100% + 1px);
  height: 1px;
  background-color: #005cb9;
}

.emotion-0:focus th:after,
.emotion-0:focus td:after {
  top: auto;
  bottom: -1px;
}

.emotion-0:focus th:first-child,
.emotion-0:focus td:first-child {
  box-shadow: inset 1px 0 0 #005cb9;
}

.emotion-0:focus th:last-child,
.emotion-0:focus td:last-child {
  box-shadow: inset -1px 0 0 #005cb9;
}

.emotion-0:focus th:last-child:before,
.emotion-0:focus td:last-child:before,
.emotion-0:focus th:last-child:after,
.emotion-0:focus td:last-child:after {
  width: 100%;
}

.emotion-0:focus th:after,
.emotion-0:focus td:after {
  z-index: 2;
}

<tr
  className="emotion-0 emotion-1"
>
  <th>
    cell
  </th>
</tr>
`;

exports[`Table Row Snapshots renders as expected 1`] = `
.emotion-0 th,
.emotion-0 td {
  background-color: #ffffff;
  padding: 12px 8px;
  font-size: 13px;
  border-right: 1px solid #dfe2e6;
  border-bottom: 1px solid #dfe2e6;
  height: 48px;
  box-sizing: border-box;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.emotion-0 td:last-child {
  border-right: none;
}

.emotion-0 th {
  height: 64px;
}

.emotion-0 th:last-child {
  border-right: none;
}

.emotion-0:last-child th,
.emotion-0:last-child td {
  border-bottom: none;
}

.emotion-0:hover th,
.emotion-0:hover td {
  background-color: #e8ebed;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:focus th,
.emotion-0:focus td {
  background-color: #d7eafc;
  position: relative;
}

.emotion-0:focus th:before,
.emotion-0:focus td:before,
.emotion-0:focus th:after,
.emotion-0:focus td:after {
  position: absolute;
  left: 0;
  top: -1px;
  content: "";
  width: calc(100% + 1px);
  height: 1px;
  background-color: #005cb9;
}

.emotion-0:focus th:after,
.emotion-0:focus td:after {
  top: auto;
  bottom: -1px;
}

.emotion-0:focus th:first-child,
.emotion-0:focus td:first-child {
  box-shadow: inset 1px 0 0 #005cb9;
}

.emotion-0:focus th:last-child,
.emotion-0:focus td:last-child {
  box-shadow: inset -1px 0 0 #005cb9;
}

.emotion-0:focus th:last-child:before,
.emotion-0:focus td:last-child:before,
.emotion-0:focus th:last-child:after,
.emotion-0:focus td:last-child:after {
  width: 100%;
}

.emotion-0:focus th:after,
.emotion-0:focus td:after {
  z-index: 2;
}

<tr
  className="emotion-0 emotion-1"
>
  <td>
    cell
  </td>
</tr>
`;

exports[`Table Row Snapshots states Alert 1`] = `
.emotion-0 th,
.emotion-0 td {
  background-color: #ffffff;
  padding: 12px 8px;
  font-size: 13px;
  border-right: 1px solid #dfe2e6;
  border-bottom: 1px solid #dfe2e6;
  height: 48px;
  box-sizing: border-box;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.emotion-0 td:last-child {
  border-right: none;
}

.emotion-0 th {
  height: 64px;
}

.emotion-0 th:last-child {
  border-right: none;
}

.emotion-0:last-child th,
.emotion-0:last-child td {
  border-bottom: none;
}

.emotion-0 th,
.emotion-0 td {
  background-color: rgba(255,161,38,0.2);
  position: relative;
  border-left-color: #fcd49f;
  border-right-color: #fcd49f;
}

.emotion-0 th:before,
.emotion-0 td:before,
.emotion-0 th:after,
.emotion-0 td:after {
  position: absolute;
  left: 0;
  top: -1px;
  content: "";
  width: calc(100% + 1px);
  height: 1px;
  background-color: #ffa126;
}

.emotion-0 th:after,
.emotion-0 td:after {
  top: auto;
  bottom: -1px;
}

.emotion-0 th:first-child,
.emotion-0 td:first-child {
  box-shadow: inset 1px 0 0 #ffa126;
}

.emotion-0 th:last-child,
.emotion-0 td:last-child {
  box-shadow: inset -1px 0 0 #ffa126;
}

.emotion-0 th:last-child:before,
.emotion-0 td:last-child:before,
.emotion-0 th:last-child:after,
.emotion-0 td:last-child:after {
  width: 100%;
}

.emotion-0:hover th,
.emotion-0:hover td {
  background-color: rgba(255,161,38,0.2);
}

<tr
  className="emotion-0 emotion-1"
>
  <td>
    cell
  </td>
</tr>
`;

exports[`Table Row Snapshots states Error 1`] = `
.emotion-0 th,
.emotion-0 td {
  background-color: #ffffff;
  padding: 12px 8px;
  font-size: 13px;
  border-right: 1px solid #dfe2e6;
  border-bottom: 1px solid #dfe2e6;
  height: 48px;
  box-sizing: border-box;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.emotion-0 td:last-child {
  border-right: none;
}

.emotion-0 th {
  height: 64px;
}

.emotion-0 th:last-child {
  border-right: none;
}

.emotion-0:last-child th,
.emotion-0:last-child td {
  border-bottom: none;
}

.emotion-0 th,
.emotion-0 td {
  background-color: rgba(255,83,71,0.2);
  position: relative;
  border-left-color: #fcc9c5;
  border-right-color: #fcc9c5;
}

.emotion-0 th:before,
.emotion-0 td:before,
.emotion-0 th:after,
.emotion-0 td:after {
  position: absolute;
  left: 0;
  top: -1px;
  content: "";
  width: calc(100% + 1px);
  height: 1px;
  background-color: #ff5347;
}

.emotion-0 th:after,
.emotion-0 td:after {
  top: auto;
  bottom: -1px;
}

.emotion-0 th:first-child,
.emotion-0 td:first-child {
  box-shadow: inset 1px 0 0 #ff5347;
}

.emotion-0 th:last-child,
.emotion-0 td:last-child {
  box-shadow: inset -1px 0 0 #ff5347;
}

.emotion-0 th:last-child:before,
.emotion-0 td:last-child:before,
.emotion-0 th:last-child:after,
.emotion-0 td:last-child:after {
  width: 100%;
}

.emotion-0:hover th,
.emotion-0:hover td {
  background-color: rgba(255,83,71,0.2);
}

<tr
  className="emotion-0 emotion-1"
>
  <td>
    cell
  </td>
</tr>
`;

exports[`Table Row Snapshots states Hover 1`] = `
.emotion-0 th,
.emotion-0 td {
  background-color: #ffffff;
  padding: 12px 8px;
  font-size: 13px;
  border-right: 1px solid #dfe2e6;
  border-bottom: 1px solid #dfe2e6;
  height: 48px;
  box-sizing: border-box;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.emotion-0 td:last-child {
  border-right: none;
}

.emotion-0 th {
  height: 64px;
}

.emotion-0 th:last-child {
  border-right: none;
}

.emotion-0:last-child th,
.emotion-0:last-child td {
  border-bottom: none;
}

.emotion-0:hover th,
.emotion-0:hover td {
  background-color: #e8ebed;
}

.emotion-0:focus {
  outline: none;
}

.emotion-0:focus th,
.emotion-0:focus td {
  background-color: #d7eafc;
  position: relative;
}

.emotion-0:focus th:before,
.emotion-0:focus td:before,
.emotion-0:focus th:after,
.emotion-0:focus td:after {
  position: absolute;
  left: 0;
  top: -1px;
  content: "";
  width: calc(100% + 1px);
  height: 1px;
  background-color: #005cb9;
}

.emotion-0:focus th:after,
.emotion-0:focus td:after {
  top: auto;
  bottom: -1px;
}

.emotion-0:focus th:first-child,
.emotion-0:focus td:first-child {
  box-shadow: inset 1px 0 0 #005cb9;
}

.emotion-0:focus th:last-child,
.emotion-0:focus td:last-child {
  box-shadow: inset -1px 0 0 #005cb9;
}

.emotion-0:focus th:last-child:before,
.emotion-0:focus td:last-child:before,
.emotion-0:focus th:last-child:after,
.emotion-0:focus td:last-child:after {
  width: 100%;
}

.emotion-0:focus th:after,
.emotion-0:focus td:after {
  z-index: 2;
}

<tr
  className="emotion-0 emotion-1"
>
  <td>
    cell
  </td>
</tr>
`;

exports[`Table Row Snapshots states Input Alert 1`] = `
.emotion-0 th,
.emotion-0 td {
  background-color: #ffffff;
  padding: 12px 8px;
  font-size: 13px;
  border-right: 1px solid #dfe2e6;
  border-bottom: 1px solid #dfe2e6;
  height: 48px;
  box-sizing: border-box;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.emotion-0 td:last-child {
  border-right: none;
}

.emotion-0 th {
  height: 64px;
}

.emotion-0 th:last-child {
  border-right: none;
}

.emotion-0:last-child th,
.emotion-0:last-child td {
  border-bottom: none;
}

.emotion-0 th,
.emotion-0 td {
  background-color: rgba(255,161,38,0.2);
}

<tr
  className="emotion-0 emotion-1"
>
  <td>
    cell
  </td>
</tr>
`;

exports[`Table Row Snapshots states Input Error 1`] = `
.emotion-0 th,
.emotion-0 td {
  background-color: #ffffff;
  padding: 12px 8px;
  font-size: 13px;
  border-right: 1px solid #dfe2e6;
  border-bottom: 1px solid #dfe2e6;
  height: 48px;
  box-sizing: border-box;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.emotion-0 td:last-child {
  border-right: none;
}

.emotion-0 th {
  height: 64px;
}

.emotion-0 th:last-child {
  border-right: none;
}

.emotion-0:last-child th,
.emotion-0:last-child td {
  border-bottom: none;
}

.emotion-0 th,
.emotion-0 td {
  background-color: rgba(255,83,71,0.2);
}

<tr
  className="emotion-0 emotion-1"
>
  <td>
    cell
  </td>
</tr>
`;

exports[`Table Row Snapshots states Selected 1`] = `
.emotion-0 th,
.emotion-0 td {
  background-color: #ffffff;
  padding: 12px 8px;
  font-size: 13px;
  border-right: 1px solid #dfe2e6;
  border-bottom: 1px solid #dfe2e6;
  height: 48px;
  box-sizing: border-box;
  -webkit-transition: background-color 0.2s;
  transition: background-color 0.2s;
}

.emotion-0 td:last-child {
  border-right: none;
}

.emotion-0 th {
  height: 64px;
}

.emotion-0 th:last-child {
  border-right: none;
}

.emotion-0:last-child th,
.emotion-0:last-child td {
  border-bottom: none;
}

.emotion-0 th,
.emotion-0 td {
  background-color: #d7eafc;
  position: relative;
}

.emotion-0 th:before,
.emotion-0 td:before,
.emotion-0 th:after,
.emotion-0 td:after {
  position: absolute;
  left: 0;
  top: -1px;
  content: "";
  width: calc(100% + 1px);
  height: 1px;
  background-color: #005cb9;
}

.emotion-0 th:after,
.emotion-0 td:after {
  top: auto;
  bottom: -1px;
}

.emotion-0 th:first-child,
.emotion-0 td:first-child {
  box-shadow: inset 1px 0 0 #005cb9;
}

.emotion-0 th:last-child,
.emotion-0 td:last-child {
  box-shadow: inset -1px 0 0 #005cb9;
}

.emotion-0 th:last-child:before,
.emotion-0 td:last-child:before,
.emotion-0 th:last-child:after,
.emotion-0 td:last-child:after {
  width: 100%;
}

.emotion-0 th:after,
.emotion-0 td:after {
  z-index: 2;
}

<tr
  className="emotion-0 emotion-1"
>
  <td>
    cell
  </td>
</tr>
`;
